From a75ef8b444db33ae0b2350e38dc0700e072b2884 Mon Sep 17 00:00:00 2001
From: Nikolay Sivov <nsivov@codeweavers.com>
Date: Fri, 21 Mar 2025 18:17:59 +0100
Subject: [PATCH] windowscodecs/metadata: Always initialize handler pointer.

In case of empty block reader we currently return uninitialized pointer
that is accessed later.

Signed-off-by: Nikolay Sivov <nsivov@codeweavers.com>
---
 dlls/windowscodecs/metadataquery.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/windowscodecs/metadataquery.c b/dlls/windowscodecs/metadataquery.c
index 609ecb7ec42..8dc71906b50 100644
--- a/dlls/windowscodecs/metadataquery.c
+++ b/dlls/windowscodecs/metadataquery.c
@@ -528,8 +528,8 @@ static void parse_query_component(struct query_parser *parser)
 static HRESULT parser_set_top_level_metadata_handler(struct query_handler *query_handler,
         struct query_parser *parser)
 {
+    IWICMetadataReader *handler = NULL;
     struct query_component *comp;
-    IWICMetadataReader *handler;
     HRESULT hr;
     GUID format;
     UINT count, i, matched_index;
-- 
GitLab

